app/models/manageiq/providers/ansible_tower/provider_mixin.rb

```diff
@@ -0,0 +1,74 @@
+module ManageIQ::Providers::AnsibleTower::ProviderMixin
+  extend ActiveSupport::Concern
+
+  def self.included(klass)
+    klass.has_many :endpoints, :as => :resource, :dependent => :destroy, :autosave => true
+
+    klass.before_validation :ensure_managers
+
+    klass.validates :name, :presence => true, :uniqueness => true
+    klass.validates :url,  :presence => true
+  end
+
+  module ClassMethods
+    def raw_connect(base_url, username, password, verify_ssl)
+      require 'ansible_tower_client'
+      AnsibleTowerClient.logger = $log
+      AnsibleTowerClient::Connection.new(
+        :base_url   => base_url,
+        :username   => username,
+        :password   => password,
+        :verify_ssl => verify_ssl
+      )
+    end
+
+    def refresh_ems(provider_ids)
+      EmsRefresh.queue_refresh(Array.wrap(provider_ids).collect { |id| [base_class, id] })
+    end
```

Comment:
Not sure what is going on in here specifically. I lifted this all right from `ansible_tower/provider.rb`